From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sunu Aziz Rahayu <sunuazizrahayu@users.noreply.github.com>
Date: Fri, 26 Sep 2026 00:00:00 +0700
Subject: [PATCH] bvN4: set WITH_LINEAGE_CHARGER before inherits

Percobaan sebelumnya (flag di lineage.mk, lalu PRODUCT_PACKAGES -=)
terbukti tidak berpengaruh: rantai inherit
common_full_phone -> common_mobile dievaluasi sebelum baris-baris itu
dieksekusi, sehingga lineage_charger_animation tetap masuk dan tabrakan
dengan charger treble tidak hilang.

Satu-satunya tempat yang pasti dievaluasi paling awal adalah puncak
file product, sebelum include/inherit apa pun. Berlaku per-varian;
varian lain (bvNE/bgN4/bgNE) menyusul saat dibutuhkan.

Patch ini juga mencabut -= yang tidak efektif di lineage.mk.

---
 lineage.mk | 5 -----
 lineage_arm64_bvN4.mk | 4 ++++
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/lineage.mk b/lineage.mk
index 0000000..0000000 100644
--- a/lineage.mk
+++ b/lineage.mk
@@ -13,10 +13,5 @@ TARGET_BOOT_ANIMATION_RES := 1080
 # APN
 PRODUCT_PACKAGES += apns-conf.xml

-# Treble already provides its own charger images
-PRODUCT_PACKAGES -= \
-    lineage_charger_animation \
-    lineage_charger_animation_vendor
-
 #Oplus fingerprint
 $(call soong_config_set,surfaceflinger,udfps_lib,//hardware/oplus:libudfps_extension.oplus)
diff --git a/lineage_arm64_bvN4.mk b/lineage_arm64_bvN4.mk
index 0000000..0000000 100644
--- a/lineage_arm64_bvN4.mk
+++ b/lineage_arm64_bvN4.mk
@@ -1,3 +1,6 @@
+# Treble provides its own charger images; must precede all inherits
+WITH_LINEAGE_CHARGER := false
+
 TARGET_GAPPS_ARCH := arm64
 include build/make/target/product/aosp_arm64.mk
 $(call inherit-product, device/phh/treble/base.mk)
--
2.54.0
